asserts/account_key.go

```diff
@@ -21,9 +21,12 @@ package asserts
 
 import (
 	"fmt"
+	"regexp"
 	"time"
 )
 
+var ValidAccountKeyName = regexp.MustCompile(`^[-a-z0-9]+$`)
```

Comment:
A couple of things here:

* I think the agreed pattern was "^(?:[a-z0-9]+-?)*[a-z](?:-?[a-z0-9])*$". But even if we keep it simple here, I don't think we want a name that starts with '-'.
* Looks like we could do with some more `name` header tests related to this pattern.
* Do we need/want to expose this?